dao-contracts/src/redistribution/cspr.rs

```diff
@@ -0,0 +1,22 @@
+use casper_dao_utils::{cspr, Address};
+use casper_types::U512;
+
+use crate::{config::Configuration, refs::ContractRefs, reputation::ReputationContractInterface};
+
+pub fn redistribute_cspr_to_all_vas(to_redistribute: U512, refs: &dyn ContractRefs) {
+    let all_balances = refs.reputation_token().all_balances();
+    let total_supply = all_balances.total_supply();
+
+    for (address, balance) in all_balances.balances() {
+        let amount = to_redistribute * balance / total_supply;
+        cspr::withdraw(*address, amount);
+    }
+}
+
+pub fn redistribute_to_governance(configuration: &Configuration, payment: U512) -> U512 {
```

Comment:
```suggestion
pub fn redistribute_to_governance(amount: U512, configuration: &Configuration) -> U512 {
```
I would use the amount as the first argument and name it just `amount`.